### ISSUE TITLE
Future of Android-Week-View?

### ISSUE THREAD

**Issue by caske33** [alamkanak/Android-Week-View#137]
Android-Week-View is immense popular (over 1000 stars on github, only 291 other java-projects have that many stars) thanks to the initial work of @alamkanak.

Unfortunately, @alamkanak is really busy lately and a lot of Pull Requests (PR) remain open and issues unanswered. I try my best to answer all the issues but I also don't have the rights to close issues if they are solved, or merge PR's when a fix is found... (And quite often, I have to link to a PR that is still open)

I see 2 options for this project to continue:
- @alamkanak gives contributors write access to this repository, so we can merge PR's and close issues and stuff like that. (and preferably also access to push to maven)
- I fork the repo so we can "start over". The problem is that some people might not update to the forked version, but for all the rest of us, we can start over with a way better version of the library.

Just pinging all people who have a PR open at the moment, but anyone is welcome to discuss:
@lolobosse, @dgrant @raplider @subbuboyapati @marunjar @alorma

**Comment by subbuboyapati** [comment]
Hi @caske33,
One week back i forked this library and customizing as i required. I am ready to contribute the same with this library. I am looking forward to support.

**Comment by lolobosse** [comment]
Hey @caske33 Cool idea, I'll try to help you out, if you do something but I think the best solution would be that @alamkanak gives you the access to the repository (you're the most active from us :+1: ) because it's a popular lib and create two libs is just more confusing I guess. :confused:

**Comment by caske33** [comment]
@lolobosse Yes, but I've asked @alamkanak before and I've proposed a more open community around this project but I never got answer... And the problem is that I'm now maintaining a lot of features on seperate branches so they could eventually be merged in which makes it 
1) not useful for the rest of the world because I think there are some nice features in them
2) hard for me to merge stuff back into my repo because a lot of things have changed and I'd have to solve merge conflicts all the time.

Having two libs, really sucks. But I've got to a point where I think for the sake of the project, we need a new core maintainer. 

I'll probably maintain this fork either on the Quivr organisation (a non-profit organisation for which I work) or on a new organization specifcally for this library

**Comment by marunjar** [comment]
hi,

sounds familiar, i also forked and modified the lib for my own use.

I also would appreciate a liltte more maintenance and the best would be write access for someone else to maintain that project if @alamkanak has less time.
Maybe we should wait a little for an answer.

Hopefully it's not necessary to fork but If (should only be the very last option) i'll also support that.

**Comment by alamkanak** [comment]
Hi everybody. At first let me tell you how much happy I am to see this community grow so fast. I am thrilled. Thanks to you for that.

As @caske33 mentioned, I really have been busy for a while these days. Truly sorry about that. You may not know but I am a student at university, engineer at a company in US, creator of some of my own products and also I do freelancing occasionally.

**Comment by marunjar** [comment]
Sounds great :+1: 

You've really a lot of work to do, thanks for having time for this too.
If i could help somehow, please feel free and say something. 


And do you have any additional info for your future plans? Cant wait for it :)

**Comment by fourier** [comment]
Great idea! Thank you for the great component you've built.

**Comment by piotrwojtowicz** [comment]
Sounds good :)

**Comment by dgrant** [comment]
Great idea!

**Comment by caske33** [comment]
Cool, super to hear that @alamkanak!

**Comment by caske33** [comment]
@alamkanak We're now almost a month and a half down the road and nothing really happened... :(

I really hope you're still in on your plan.

**Comment by caske33** [comment]
This has now been fixed. I've got collaborator rights and I'm trying my best to keep up with the library. Also big thanks to @marunjar for helping me. If anybody else should get collaborator rights, it should be him.

As for the roadmap, I think the next 3 things are the most important to focus on:
1. Custom layout for the events (serveral related issues).
2. Limit the dates which you can scroll (#6)
3. Rendering Full-day events #101

**Comment by entropitor** [comment]
I'm currently slammed as well and just like last year, there are again a large number of open Pull Requests (PR's). That's why I think it might be a good idea to have a more open community around this library. About a year ago we had 1000 stars, now we have 1620. So people really love this. I think we also have a bunch of people who want to contribute, but we're just lacking the people to look at the PR's and provide them the feedback they need. 

This project only has @alamkanak and me as collaborator (we can push to the master branch, @alamkanak is the only one who can publish to maven). Other projects with a more open collaborator policy have shown that a more open policy can work (for example if someone who makes a substantial PR, gets collaborator rights after it is merged. If they don't behave, they lose the rights again. A smaller group of people has admin rights. They can all publish to maven). Usually the amount of work that gets done raises a lot.

Within 1 year, I'll be gone from my current job where I use the library. @jhoobergs has mainly taken over my job within that organization. So within 1 year, I'll probably be completly inactive in this project and stop watching the repository. 

I think it's time we start planning a sustainable future for this project. Before it's too late...

@alamkanak @lolobosse @marunjar What do you think? How should we proceed?

**Comment by jhoobergs** [comment]
I think @caske33 is right, this library has still a lot of potential but a lot of work is thrown away because nobody takes care about the PR's. I think everyone would benefit from a more open community.

**Comment by thelittlefireman** [comment]
i'm volunteer to fork this repo and work on it hardly again 
@caske33

**Comment by entropitor** [comment]
Should we fork under an organization with the name AndroidWeekView or something like that? @jhoobergs @thelittlefireman

**Comment by marioguerriero** [comment]
Hi, I read your discussion and I  would appreciate it you would fork this repository and keep working on new features.

**Comment by entropitor** [comment]
@jhoobergs and I have forked the repo at [Quivr/Android-Week-View](https://github.com/Quivr/Android-Week-View) where we will try to have a more open contribution model.

**Comment by entropitor** [comment]
(Most open pull requests from this repository are merged in that repository)
